pkg/operator/controller/dns/controller.go

```diff
@@ -160,14 +161,6 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
-	// Only process records associated with ingresscontrollers, because this isn't
-	// intended to be a general purpose DNS management system.
-	ingressName, ok := record.Labels[manifests.OwningIngressControllerLabel]
```

Comment:
I wonder if instead of removing this entirely, we could add a label for gateway DNS records that we need.  Then, we are only managing 2 known kinds of records instead of becoming a general purpose DNS management system.

Reply:
We could, but if people wanted to abuse this controller, they could just as easily create a dnsrecord CR with whatever label we wrote the controller to check for.  Checking labels seems like unnecessary complexity with little benefit.